atc/scheduler/factory/factory_parallel_test.go

```diff
@@ -0,0 +1,210 @@
+				MaxInParallel: 1,
+				Steps: []atc.Plan{
+					expectedPlanFactory.NewPlan(atc.TaskPlan{
+						Name:                   "some thing",
+						VersionedResourceTypes: resourceTypes,
+					}),
+					expectedPlanFactory.NewPlan(atc.TaskPlan{
+						Name:                   "some other thing",
+						VersionedResourceTypes: resourceTypes,
+					}),
+				},
+			})
+			Expect(actual).To(Equal(expected))
+		})
+	})
+
+	Context("when I have nested parallel steps", func() {
+		It("returns the correct plan", func() {
+			actual, err := buildFactory.Create(atc.JobConfig{
+				Plan: atc.PlanSequence{
+					{
+						Parallel: &atc.PlanSequence{
+							{
+								Task: "some thing",
+							},
+							{
+								Parallel: &atc.PlanSequence{
+									{
+										Task: "some nested thing",
+									},
+									{
+										Task: "some nested other thing",
+									},
+								},
+								MaxInParallel: 2,
+							},
+						},
+						MaxInParallel: 1,
+					},
+				},
+			}, resources, resourceTypes, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			expected := expectedPlanFactory.NewPlan(atc.ParallelPlan{
+				MaxInParallel: 1,
+				Steps: []atc.Plan{
+					expectedPlanFactory.NewPlan(atc.TaskPlan{
+						Name:                   "some thing",
+						VersionedResourceTypes: resourceTypes,
+					}),
+					expectedPlanFactory.NewPlan(atc.ParallelPlan{
+						MaxInParallel: 2,
+						Steps: []atc.Plan{
+							expectedPlanFactory.NewPlan(atc.TaskPlan{
+								Name:                   "some nested thing",
+								VersionedResourceTypes: resourceTypes,
+							}),
+							expectedPlanFactory.NewPlan(atc.TaskPlan{
+								Name:                   "some nested other thing",
+								VersionedResourceTypes: resourceTypes,
+							}),
+						},
+					}),
+				},
+			})
+			Expect(actual).To(Equal(expected))
+		})
+	})
+
+	Context("when I have a parallel with hooks", func() {
+		It("returns the correct plan", func() {
+			actual, err := buildFactory.Create(atc.JobConfig{
+				Plan: atc.PlanSequence{
+					{
+						Parallel: &atc.PlanSequence{
+							{
+								Task: "some thing",
+								Success: &atc.PlanConfig{
+									Task: "some success hook",
+								},
+							},
+						},
+						MaxInParallel: 1,
+					},
+				},
+			}, resources, resourceTypes, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			expected := expectedPlanFactory.NewPlan(atc.ParallelPlan{
+				Steps: []atc.Plan{
+					expectedPlanFactory.NewPlan(atc.OnSuccessPlan{
+						Step: expectedPlanFactory.NewPlan(atc.TaskPlan{
+							Name:                   "some thing",
+							VersionedResourceTypes: resourceTypes,
+						}),
+						Next: expectedPlanFactory.NewPlan(atc.TaskPlan{
+							Name:                   "some success hook",
+							VersionedResourceTypes: resourceTypes,
+						}),
+					}),
+				},
+				MaxInParallel: 1,
+			})
+			Expect(actual).To(Equal(expected))
+		})
+	})
+
+	Context("when I have a hook on a parallel", func() {
```

Comment:
Delete? If hooks work for any other step type, it should work for Parallel?

Reply:
Agreed